lib/rabbitmq_message_deduplication/cache_manager.ex

```diff
@@ -100,6 +100,7 @@ defmodule RabbitMQMessageDeduplication.CacheManager do
       {:timeout, reason} -> {:error, reason}
       error -> error
     end
+    :net_kernel.monitor_nodes(true)
```

Comment:
Elixir is an expression-based functional language: everything is an expression and, therefore, returns a value.

When you write a function, you need to keep that into consideration. The last expression will determine what the function returns. In this case, you are returning either `:ok` or any error information related to `net_kernel.monitor_nodes`. You are effectively hiding any other possible error which might have happened before (table creation and initialization failures). Moreover, you are going to always monitor nodes even in case of catastrophic failure in initializing the cache manager.

The [`gen_server.init`](https://hexdocs.pm/elixir/1.12/GenServer.html#c:init/1) function shall return a tuple `{:ok, state}` in case of success to notify its supervisor process that everything is ok. The `state` in particular is used to maintain any information that the `gen_server` might need.

I would suggest to write it as follows.

```elixir
  alias :timer, as: Timer
  alias :mnesia, as: Mnesia
  alias :net_kernel, as: NetKernel    # Alias Erlang OTP modules to make it easier to use in your code
  alias RabbitMQMessageDeduplication.Cache, as: Cache

  ...

    with :ok <- mnesia_create(Mnesia.create_table(@caches, [])),
         :ok <- mnesia_create(Mnesia.add_table_copy(@caches, node(), :ram_copies)),
         :ok <- Mnesia.wait_for_tables([@caches], @cache_wait_time), 
         :ok <- NetKernel.monitor_nodes(true)
    do
      Process.send_after(__MODULE__, :cleanup, @cleanup_period)
      {:ok, state}
    else
      {:timeout, reason} -> {:error, reason}
      error -> error
    end
```

In this way, you leverage the [`with`](https://elixirschool.com/en/lessons/basics/control_structures#with-3) special form to better take into account issues and return them to the supervising process for logging in case of error.